src/vs/platform/storage/electron-main/storageIpc.ts

```diff
@@ -156,3 +158,95 @@ export class StorageDatabaseChannel extends Disposable implements IServerChannel
 		return storage;
 	}
 }
+
+export class ProfileStorageChangesListenerChannel extends Disposable implements IServerChannel {
```

Comment:
Why do we need an event for profile changes in storage?

Reply:
Settings Sync (Global State) needs to know when there are changes to any profile storage so that it can sync. StorageService is in shared process only knows about default profile. `IProfileStorageService` emits an event when any profile storage data is changed (incl default).